osquery/core/windows/wmi.cpp

```diff
@@ -349,20 +349,40 @@ WmiRequest::WmiRequest(const std::string& query, BSTR nspace) {
   locator_.reset(locator);
 
   IWbemServices* services = nullptr;
+  BSTR nspace_str = SysAllocString(nspace.c_str());
+  if (NULL == nspace_str) {
+    return;
+  }
+
   hr = locator_->ConnectServer(
-      nspace, nullptr, nullptr, nullptr, 0, nullptr, nullptr, &services);
+      nspace_str, nullptr, nullptr, nullptr, 0, nullptr, nullptr, &services);
+  SysFreeString(nspace_str);
+
   if (hr != S_OK) {
     return;
   }
   services_.reset(services);
 
   IEnumWbemClassObject* wbem_enum = nullptr;
 
-  hr = services_->ExecQuery((BSTR)L"WQL",
-                            (BSTR)wql.c_str(),
+  BSTR language_str = SysAllocString(L"WQL");
+  if (NULL == language_str) {
+    return;
+  }
+
+  BSTR wql_str = SysAllocString(wql.c_str());
+  if (NULL == wql_str) {
+    return;
```

Comment:
If the code returns early here then `language_str` is leaked, we will need to call `SysFreeString(language_str)`

Reply:
Good catch! Maybe later we can use _bstr_t to auto-destruct these.